cfdtools/data.py

```diff
@@ -186,3 +184,37 @@ def _dumphdfgroup(self, hgroup: hdf5.Group, **options):
             datagroup = hgroup.create_group(f"i{i:06}")
             for vname, var in datadict.items():
                 datagroup.create_dataset(vname, data=var, **options)
+
+    def xdmf_content(self, filename, geometry_content):
+        """Create the XMF content associated to the data set.
+
+        :param str filename: Name of the output HDF5 file.
+        :param list(str) geometry_content: XDMF content for the geometry and topology of the mesh.
+          The mesh is time-invariant.
+        :return: The XMF content.
+        :rtype: list(str)
+        """
+        lines = ['<Grid Name="IC3" GridType="Collection" CollectionType="Temporal">']
+
+        for i, datadict in enumerate(self._datalist):
+            lines += ['<Grid Name="Unstructured Mesh">']
+            lines += [f'<Time Value="{i}"/>']
+            lines += geometry_content
+            for vname, var in datadict.items():
+                if len(var.shape) == 1:
```

Comment:
Shouldn't it be 0 for scalars and 1 for vectors (and 2 for tensors) ?
`len(numpy.array([1,2])[0].shape)`
Or comment to explain the "+1" ?